client/src/styles/chat.css

```diff
@@ -52,6 +52,11 @@
         @apply bg-red p-3 break-all will-change-auto flex gap-6 items-center;
     }
 
+    .content--reply * {
+        margin: 0px;
+        padding: 0px;
+    }
+
```

Comment:
it's not, it's actually here

Reply:
> it's not, it's actually here

Sorry, I don't get you

Reply:
but then again, this is going to cause a huge conflict, what i understand this issue to fix is click on the markdown to point to the message it's replying to and not style it, so why are you styling it?

Reply:
also there's a new guideline for if you work on the server side, you should send a screen recording of the working software

Reply:
> but then again, this is going to cause a huge conflict, what i understand this issue to fix is click on the markdown to point to the message it's replying to and not style it, so why are you styling it?

When a message is a reply to a previous message, a preview of the previous message is shown right? Yes.

But in the preview, we're only trying to display part (a preview) of the original message. But the original message contains markdown content with it's own styling (e.g `p+p` has a padding), and all that does not make the preview work as expected. The only option is to reset the default styling (especially the ones causing issues) of the markdown content, which is what I did.

Reply:
> it's not, it's actually here, if that's what you wanted to do, which i understand now, markdown by default has no styles
> 
> https://github.com/Dun-sin/Whisper/blob/c7c0e4ddafcf7babe12d2f22ccd3ccbdfb2f1ec2/client/src/styles/chat.css#L90-L92

@mathiasayivor i think you missed my updated comment and as i said it's going to cause a huge conflict as there's an issue to style the QR properly

Reply:
what's the difference between a quote reply and a reply, do you understand the conflict here now?

Reply:
you need to wrap your code around the existing one, this is causing a problem

Reply:
> you need to wrap your code around the existing one, this is causing a problem

At this point, are you referring to the whole pr, or just the CSS part?

Reply:
both

Reply:
> what's the difference between a quote reply and a reply, do you understand the conflict here now?


A quote reply doesn't know the original message being replied.

Reply:
here's a summary incase you are confused:
- adding a `reply` when there's a `quote reply` option is a contradiction and is going to cause problems, you have to wrap your new code around the existing functionality. Which isn't what you did
- by default the markdown isn't styled i already gave you where it was styled so there's actually nothing to `reset` like you thought, remember tailwind removes the default styling for everything

Reply:
lastly keep track of #346

Reply:
> here's a summary incase you are confused:
> 
>     * adding a `reply` when there's a `quote reply` option is a contradiction and is going to cause problems, you have to wrap your new code around the existing functionality. Which isn't what you did
> 
>     * by default the markdown isn't styled i already gave you where it was styled so there's actually nothing to `reset` like you thought, remember tailwind removes the default styling for everything

I disagree with your last point. MarkdownIt applies some base styles after parsing the markdown syntax to make it appear as expected.

Reply:
i was the one who made the original change, for the quote reply, i thought it wasn't working because i expected what you just talked about, until i realized i had to style it myself

Reply:
it's the only exception to the rule, i want to remove all applies until it's absolutely necessary i need to do that, i thought yours wasn't because i assumed you saw the blockquote style and was just making another

Reply:
Ok. So do you get my reason now?

Reply:
huh?

Reply:
🤣 Are you also confused as me?

Reply:
![Screenshot from 2023-10-16 20-01-16](https://github.com/Dun-sin/Whisper/assets/74247733/a05dac8d-01e7-4a3e-8150-5a670561b971)
This is one of the styles added by the MarkdownIt library

Reply:
> 🤣 Are you also confused as me?

Oh just didn't read your reply right

Reply:
> ![Screenshot from 2023-10-16 20-01-16](https://github.com/Dun-sin/Whisper/assets/74247733/a05dac8d-01e7-4a3e-8150-5a670561b971)
> This is one of the styles added by the MarkdownIt library

Okay missed that because I never noticed any change when I used it, but the quote reply is a blockquote element, not a p. 

And to quote which is what I used is this (>) which goes back to my first point of making a new change around the existing code.

Reply:
The reason why my solution wasn't wrapped around the `quote reply` is because `quote reply` is basically just a string, and it's not possible to know the original message. Even if you use string matching, it is prone to errors because there may be multiple messages containing the quoted string.

Reply:
But you see how it doesn't make sense to have both right, so you do have to do that, wrap around the existing I mean👍🏽

Reply:
Do you have any suggestions on how to go about it?

Reply:
Nope

Reply:
@Dun-sin where can I locate the code for displaying the Quoted reply? Is it just CSS that's used to style the quoted reply?

Reply:
https://github.com/Dun-sin/Whisper/blob/f3119aac9e7ddc225869a12b71aa6983fccc8f91/client/src/components/Chat.jsx#L324

Reply:
please keep an eye on this PR -> #456

Reply:
best choice

Reply:
@mathiasayivor it's closed

Reply:
> @mathiasayivor it's closed

#346 is still open

Reply:
@mathiasayivor closed now

Reply:
@Dun-sin the quote reply function doesn't seem to be working. Is that intentional, or a bug?

Reply:
Just remove it and replace it with your own implementation, my issue is they just can't be both there